routes/booking_test.go

```diff
@@ -0,0 +1,422 @@
+	t := suite.T()
+	rr := executeReq(t, &testRouteConfig{
+		Method:  http.MethodGet,
+		Body:    nil,
+		Handler: suite.app.GetBookingsByWorkspaceID,
+		URL:     fmt.Sprintf("/bookings/workspaces/%s", "2"),
+		URLParams: map[string]string{
+			"workspace_id": "2",
+		},
+	})
+
+	assert.Equal(t, http.StatusInternalServerError, rr.Code, "status code")
+}
+
+func (suite *AppTestSuite) TestGetBookingsByWorkspaceID() {
+	bookingId := Booking1.WorkspaceID
+	t := suite.T()
+	rr := executeReq(t, &testRouteConfig{
+		Method:  http.MethodGet,
+		Body:    nil,
+		Handler: suite.app.GetBookingsByWorkspaceID,
+		URL:     fmt.Sprintf("/bookings/workspaces/%s", bookingId),
+		URLParams: map[string]string{
+			"workspace_id": bookingId,
+		},
+	})
+
+	assert.Equal(t, rr.Code, http.StatusOK, "status code")
+
+	var payload []*model.Booking
+	_ = json.Unmarshal(rr.Body.Bytes(), &payload)
+	assert.Equal(t, 2, len(payload), "testGetBookingsByWorkspaceID: incorrect response size")
+	expectedBookings := []*model.Booking{Booking1, Booking2}
+	for _, expected := range expectedBookings {
+		found := false
+		for _, b := range payload {
+			if b.Equal(expected) {
+				found = true
+			}
+		}
+		if !found {
+			t.Fatalf("testGetBookingsByWorkspaceID: %s not found in offring list", expected.ID)
+		}
+	}
+}
+
+func (suite *AppTestSuite) TestGetOneBookingByUserIDFail() {
+	t := suite.T()
+	rr := executeReq(t, &testRouteConfig{
+		Method:  http.MethodGet,
+		Body:    nil,
+		Handler: suite.app.GetBookingsByUserID,
+		URL:     fmt.Sprintf("/bookings/users/%s", "13"),
+		URLParams: map[string]string{
+			"user_id": "2",
+		},
+	})
+
+	assert.Equal(t, http.StatusInternalServerError, rr.Code, "status code")
+}
+
+func (suite *AppTestSuite) TestGetBookingsByUserID() {
+	bookingId := Booking1.UserID
+	t := suite.T()
+	rr := executeReq(t, &testRouteConfig{
+		Method:  http.MethodGet,
+		Body:    nil,
+		Handler: suite.app.GetBookingsByUserID,
+		URL:     fmt.Sprintf("/bookings/users/%s", bookingId),
+		URLParams: map[string]string{
+			"user_id": bookingId,
+		},
+	})
+	assert.Equal(t, rr.Code, http.StatusOK, "status code")
+	var payload []*model.Booking
+	_ = json.Unmarshal(rr.Body.Bytes(), &payload)
+	assert.Equal(t, 13, len(payload), "testGetBookingsByUserID: incorrect response size")
+	expectedBookings := []*model.Booking{Booking1, Booking2}
+	for _, expected := range expectedBookings {
+		found := false
+		for _, b := range payload {
+			if b.Equal(expected) {
+				found = true
+			}
+		}
+		if !found {
+			t.Fatalf("testGetBookingsByUserID: %s not found in offring list", expected.ID)
+		}
+	}
+}
+
+func (suite *AppTestSuite) TestGetBookingsByDateRange() {
+	bookingStart := "1548547200"
+	bookingEnd := "1548719999"
+	t := suite.T()
+	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("/bookings?%s&%s", bookingStart, bookingEnd), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	q := req.URL.Query()
+	q.Add("start", bookingStart)
```

Comment:
i believe u dont need to do this, just adding the params to the URL in `http.NewRequest` should work fine. 
I have an example where i used the `executeReq()` to do it https://github.com/iwork-cs319/iwork-backend/blob/30535c8687fbec5ac1c5e8eae2ffa75bd5232d7c/routes/workspace_test.go#L63

Reply:
if its possible to use the execRequest, try using it. Being nitpicky so that all the tests are uniform. Feel free to change the executeReq function to add more functionality if needed

Reply:
Yea for some reason when I did this using executeReq() with URLParams it did not work. This is why I had to add this in. I'll try testing this in a bit.

Reply:
yea urlParams are for the route matching `/workspaces/:id`
so `URLParams: map[string]string{ "id": "2"}` but for queryParams u can just have it in the url itself and dont need anything else. Does that make sense?

Reply:
Works now. Must have been some formatting or something.. thanks!